src/Native/System.Native/pal_networking.cpp

```diff
@@ -1215,9 +1223,14 @@ extern "C" Error SystemNative_SetIPv4MulticastOption(int32_t socket, int32_t mul
         return PAL_EINVAL;
     }
 
+#if HAVE_IP_MREQN
     ip_mreqn opt = {.imr_multiaddr = {.s_addr = option->MulticastAddress},
                     .imr_address = {.s_addr = option->LocalAddress},
                     .imr_ifindex = option->InterfaceIndex};
+#else
+    ip_mreq opt = {.imr_multiaddr = {.s_addr = option->MulticastAddress},
+                   .imr_interface = {.s_addr = option->LocalAddress}};
+#endif
```

Comment:
The question is really about whether it's ok to ignore the managed `MulticastOption.InterfaceIndex` on systems that don't support it.  Your change fixes this for the systems that *do* support it; but on those that don't, maybe we should be throwing `PlatformNotSupportedException` from `Socket.SetSocketOption.`

OTOH, that's an existing problem, and your change isn't making it worse, so maybe we should just log a separate issue for this.